src/common/commandExecutor.ts

```diff
@@ -74,29 +87,18 @@ export class CommandExecutor {
         return this.spawnReactCommand("start", args, options);
     }
 
-    /**
-     * Uses the `react-native -v` command to get the version used on the project.
-     * Returns null if the workspace is not a react native project
-     */
     public getReactNativeVersion(): Q.Promise<string> {
-        let deferred = Q.defer<string>();
-        const reactCommand = HostPlatform.getNpmCliCommand(CommandExecutor.ReactNativeCommand);
-        let output = "";
-
-        const result = this.childProcess.spawn(reactCommand,
-            [CommandExecutor.ReactNativeVersionCommand],
-            { cwd: this.currentWorkingDirectory });
-
-        result.stdout.on("data", (data: Buffer) => {
-            output += data.toString();
-        });
-
-        result.stdout.on("end", () => {
-            const match = output.match(/react-native: ([\d\.]+)/);
-            deferred.resolve(match && match[1] || "");
-        });
-
-        return deferred.promise;
+        let curPackage = new Package(this.currentWorkingDirectory);
+        return curPackage.dependencyPackage("react-native").version()
+            .catch(err => {
+                return curPackage.dependencies()
+                    .then(dependencies => {
+                        if (dependencies["react-native"] && dependencies["react-native"].match(/[\d\.]+/)) {
```

Comment:
A version validation is implemented in `isSupportedVersion` method in [`rn-extension`](https://github.com/RedMickey/vscode-react-native/blob/a4bcfb3528de1a1e4e2222e5ce494dfd436f4521/src/extension/rn-extension.ts#L213) file. Here we only take the version of React Native and check its matching to version's template. Can it be better to remove this `match` validation, because there is another in `rn-extension`?